File: android/app/src/main/kotlin/com/goalmaker/app/domain/backup/BackupDocument.kt
package com.goalmaker.app.domain.backup

import kotlinx.serialization.json.JsonObject

/**
 * One export: who it belongs to, what wrote it, and every row it carries, table by table
 * (docs/backup.md). Tombstones are left out, so a row may point at something the file does not
 * carry.
 */
data class BackupDocument(
    val version: Int,
    val exportedAt: String,
    val app: String,
    val appVersion: String,
    val owner: String,
    /** Rows by table, in the order a restore applies them. */
    val tables: Map<String, List<JsonObject>>,
) {
    val rowCount: Int get() = tables.values.sumOf { it.size }
}

File: windows/src/GoalMaker.Core/Backup/BackupDocument.cs
using System.Text.Json.Nodes;

namespace GoalMaker.Core.Backup;

/// <summary>
/// One export: who it belongs to, what wrote it, and every row it carries, table by table
/// (docs/backup.md). Tombstones are left out, so a row may point at something the file does not
/// carry.
/// </summary>
public sealed record BackupDocument(
    int Version,
    string ExportedAt,
    string App,
    string AppVersion,
    string Owner,
    IReadOnlyDictionary<string, IReadOnlyList<JsonObject>> Tables)
{
    public int RowCount => Tables.Values.Sum(rows => rows.Count);
}
